src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -867,8 +867,7 @@ Copyright (c) .NET Foundation. All rights reserved.
          generate a different one. -->
     <PropertyGroup>
       <_TrimRuntimeAssets Condition="'$(PublishSingleFile)' == 'true' and '$(SelfContained)' == 'true'">true</_TrimRuntimeAssets>
-      <_UseBuildDependencyFile Condition="'@(_ExcludeFromPublishPackageReference)' == '' and
-                                          '@(RuntimeStorePackages)' == '' and
+      <_UseBuildDependencyFile Condition="'@(RuntimeStorePackages)' == '' and
                                           '$(PreserveStoreLayout)' != 'true' and
```

Comment:
I think this change is fine, I'm just curious about the usage of the `_UseBuildDependencyFile` on line 539:
https://github.com/dotnet/sdk/blob/4e27ad22e1513ce8920f7059daf004235cd516fc/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L539-L542

We're making a decision if it's OK to keep the `.deps.json` from the build, but that really says nothing about the apphost, there are cases where we will need a new apphost anyway (single-file) and there are cases where we need to rebuild the `.deps.json` but could keep the apphost as-is (trimming).

I don't think this change will have any real effect on that though, so it's really unrelated.